src/components/ThemeToggler/ThemeToggler.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+
+ThemeToggler.propTypes = {
+  theme: PropTypes.string,
+  toggleTheme: PropTypes.func,
+}
+
+export default function ThemeToggler({ theme, toggleTheme }) {
+  const isDarkMode = theme === 'dark' ? true : false
+
+  return (
+    <>
+      <ToggleWrapper>
+        <input
+          type="checkbox"
+          id="themeToggle"
+          onClick={toggleTheme}
+          defaultChecked={theme === 'dark' ? true : false}
+        />
+        <ToggleLabel htmlFor="themeToggle" isDarkMode={isDarkMode}>
+          <ToggleHandler isDarkMode={isDarkMode}>
+            <Crater isDarkMode={isDarkMode} />
```

Comment:
Crater würde ich in eine Craters Komponente wrappen. Ähnliches für Star in Stars.